### DISCUSSION TITLE
How can `dropShadows` be defined as a TokenCategory?

### DISCUSSION THREAD

**Discussion by k35o** [chakra-ui/panda#2677] [Q&A]
Hi I want to use the `filter` property in panda-css.

According to [the documentation](https://panda-css.com/docs/utilities/effects#filter), if I set `filter` to `'auto'`, the `filter` will be `'var(--blur, ) var(--brightness, ) var(--contrast, ) var(--grayscale, ) var(--hue- rotate, ) var(--invert, ) var(--saturate, ) var(--sepia, ) var(--drop-shadow, )'`.
`var(--drop-shadow, )` is set to the value given to `dropShadow`, and the value to be given is written to make use of `dropShadows` in the Token Category.

| Prop         | CSS Property    | Token Category |
| ------------ | --------------- | -------------- |
| `filter`     | `filter`        | none           |
| `blur`       | `--blur`        | `blurs`        |
| `brightness` | `--brightness`  | none           |
| `contrast`   | `--contrast`    | none           |
| `grayscale`  | `--grayscale`   | none           |
| `hueRotate`  | `--hue-rotate`  | none           |
| `invert`     | `--invert`      | none           |
| `saturate`   | `--saturate`    | none           |
| `sepia`      | `--sepia`       | none           |
| `dropShadow` | `--drop-shadow` | `dropShadows`  |

However, the `dropShadows` does not appear except here.
If I want to use `dropShadows` as a token, how should I describe the configuration file?

**Comment by anubra266** [comment]
@k35o This is how you'd use it. Seems we need to update the types for the config though.
https://play.panda-css.com/IdIUCE-bGR

**Reply by k35o** [reply]
@anubra266 
Thank you. Does this mean that it is not supported as a type?

Could necessary changes just be added here?
https://github.com/chakra-ui/panda/blob/adb2484c19e3ff89baddbbad84cb22e89d81b69c/packages/types/src/tokens.ts#L69

**Reply by anubra266** [reply]
Yes you're correct. There's probably other places it needs to be added too

**Reply by k35o** [reply]
Can I set up a PR with reference to [this commit](https://github.com/chakra-ui/panda/commit/e2f9bb367e1fc10510a3f048275d7340be1dfbd7)?

**Reply by anubra266** [reply]
Yes you can @k35o

**Reply by anubra266** [reply]
@k35o I think instead of accepting only strings we should accept an object just like shadow.

Then see this for transform reference:

https://github.com/chakra-ui/panda/blob/adb2484c19e3ff89baddbbad84cb22e89d81b69c/packages/token-dictionary/src/transform.ts#L14

This is the only difference to keep in mind:

![CleanShot 2024-06-19 at 02 37 07@2x](https://github.com/chakra-ui/panda/assets/30869823/633c92f8-5713-44c8-92f5-fec7fdf6cc62)

**Reply by k35o** [reply]
dropShadows can be added to Tokens and accept array and object like shadow.
https://github.com/chakra-ui/panda/pull/2678

**Reply by k35o** [reply]
@anubra266
https://github.com/chakra-ui/panda/pull/2681 seems to have been Closed, what is your policy on this?

**Reply by anubra266** [reply]
@k35o  I’d recommend that you add the drop shadow tokens to your utilities for now.
In our end, we’ll remove dropShadows from the preset.
